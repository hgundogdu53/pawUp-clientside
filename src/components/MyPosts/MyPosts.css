.myPost-container {
    background-color: rgba(93, 136, 168, 0.616);
    transition: 0.3s;
    border-radius: 5px;
    width: 22.5rem;
    margin-top: 0.5rem;
    padding: 15px 10px;
}

.myPost-container:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}


.container {
    max-height: 379.5px;
}

.deletePostBtn,
.update-btn {
    background-color: #fff;
    color: #004690;
    margin-top: 15px;
    padding: 7.5px 20px;
    margin-left: 0.5rem;
    text-transform: uppercase;
    text-decoration: none;
    font-weight: bold;
    text-decoration: none;
    display: inline-block;
    transition: all .2s;
    border-radius: 100px;
}

.deletePostBtn:hover,
.update-btn:hover {
    cursor: pointer;
    transform: translateY(-2.5px);
    box-shadow: 0 5px 15px rgba(0, 17, 255, 0.2);
}

.deletePostBtn:active,
.update-btn:active {
    transform: translateY(-1px);
}



@media only screen and (min-width: 600px) {
    .myPost-grid {
        display: grid;
        grid-gap: 0.5rem 0.75rem;
        grid-template-columns: auto auto;
        justify-content: space-evenly;
        overflow: hidden;
        grid-template-areas: 'myArea myArea';
    }
}

@media only screen and (min-width: 1200px) {
  .myPost-grid {
      display: grid;
      grid-gap: 0.5rem 0.25rem;
      grid-template-columns: auto auto auto;
      justify-content: space-evenly;
      overflow: hidden;
      grid-template-areas: 'myArea myArea myArea';
  }
}