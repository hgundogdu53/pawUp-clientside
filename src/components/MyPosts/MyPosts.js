import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import config from '../../config';
import TokenService from '../../services/token-service';
import './MyPosts.css';


class MyPosts extends Component {
    constructor(props) {
        super(props);
        this.state = {
            posts: []
        }
    }

    componentDidMount() {
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        }
        fetch(`${config.API_ENDPOINT}/posts/myPosts`, options)
            .then(res => {
                if (res.ok) {
                    return res.json();
                }
                else {
                    throw new Error('Something went wrong');
                }
            })
            .then(data => {
                this.setState({ posts: data })
            })
            .catch(err => {
                alert(`Something went wrong ${err}`)
            })
    }

    render() {
        return (
            <>
                <main className='flex-container'>
                    <header role="banner">
                        <h1>My Posts</h1>
                    </header>
                    <section className='myPost-grid'>
                        {this.state.posts.map(post => {
                            return (
                                <section key={post.id} className='myPost-container'>
                                    <h4 className='dog_name'>{post.dog_name}</h4>
                                    <section className='container'>
                                        <p>Owner Email:  <a href={`mailto:${post.email}?subject=Interested in ${post.pet_name}!`} className='email'>{post.email}</a></p>
                                        <p>Birthdate: {moment(post.birthdate).format("MM-DD-YYYY")}</p>
                                        <p>Types-of-pet: {post.types_of_pet}</p>
                                        <p>Hobbies: {post.hobbies}</p>
                                    </section>
                                    <button className='deletePostBtn' onClick={() => (
                                        /* eslint-disable */
                                        this.props.handleDeletePost(post.id, this.handleDeletePost),
                                        this.setState({ posts: this.state.posts.filter(posts => posts.id !== post.id) })
                                        /* eslint-disable */
                                    )
                                    }>Delete</button>
                                    <Link to='/edit'><button className='update-btn' onClick={() => {
                                        this.props.setId(post.id)
                                    }}>Update</button></Link>
                                </section>
                            )
                        })}
                    </section>
                </main>
            </>
        )
    }
}


export default MyPosts;