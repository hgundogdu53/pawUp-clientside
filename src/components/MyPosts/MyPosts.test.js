import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MyPosts from './MyPosts';

describe(`MyPosts component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<MyPosts />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});