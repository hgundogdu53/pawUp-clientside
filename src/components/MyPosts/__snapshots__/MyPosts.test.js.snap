// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPosts component renders without crashing 1`] = `
<Fragment>
  <main
    className="flex-container"
  >
    <header
      role="banner"
    >
      <h1>
        My Posts
      </h1>
    </header>
    <section
      className="myPost-grid"
    />
  </main>
</Fragment>
`;
